examples/java/custom-panel/src/main/java/custompanel/CustomPanelBuilder.java

```diff
@@ -0,0 +1,333 @@
+package custompanel;
+
+import java.util.LinkedList;
+import java.util.List;
+
+import com.grafana.foundation.cog.Builder;
+import com.grafana.foundation.cog.variants.Dataquery;
+import com.grafana.foundation.dashboard.DashboardFieldConfigSourceOverrides;
+import com.grafana.foundation.dashboard.DashboardLink;
+import com.grafana.foundation.dashboard.DataSourceRef;
+import com.grafana.foundation.dashboard.DataTransformerConfig;
+import com.grafana.foundation.dashboard.FieldColor;
+import com.grafana.foundation.dashboard.FieldConfig;
+import com.grafana.foundation.dashboard.FieldConfigSource;
+import com.grafana.foundation.dashboard.GridPos;
+import com.grafana.foundation.dashboard.LibraryPanelRef;
+import com.grafana.foundation.dashboard.Panel;
+import com.grafana.foundation.dashboard.PanelRepeatDirection;
+import com.grafana.foundation.dashboard.ThresholdsConfig;
+import com.grafana.foundation.dashboard.ValueMapping;
+
+public class CustomPanelBuilder implements Builder<Panel> {
```

Comment:
I need to investigate it a bit. If I extends from `Panel.Builder` I don't have access to the "options" since its internal Panel is private. Other thing should be try to make Panel abstract or a kind of interface.